spec/models/embed/purl/resource_file_spec.rb

```diff
@@ -0,0 +1,279 @@
+      end
+
+      it 'is true for identify world accessible objects' do
+        stub_purl_response_with_fixture(file_purl)
+        expect(Embed::Purl.new('12345').contents.first.files.all?(&:world_downloadable?)).to be true
+      end
+    end
+  end
+
+  describe 'image data' do
+    before { stub_purl_response_with_fixture(image_purl) }
+
+    let(:image) { Embed::Purl.new('12345').contents.first.files.first }
+
+    it 'gets the image height and width for image objects' do
+      expect(image.height).to eq '6123'
+      expect(image.width).to eq '5321'
+    end
+  end
+
+  describe 'file data' do
+    before { stub_purl_response_with_fixture(file_purl) }
+
+    let(:file) { Embed::Purl.new('12345').contents.first.files.first }
+
+    it 'gets the location data when available' do
+      expect(file.location).to eq 'http://stacks.stanford.edu/file/druid:abc123/Title_of_the_PDF.pdf'
+    end
+  end
+
+  describe 'duration' do
+    it 'gets duration string from videoData' do
+      f = double('File')
+      video_data_el = Nokogiri::XML("<videoData duration='P0DT1H2M3S'/>").root
+      expect(f).to receive(:xpath).with('./*/@duration').and_return(['something'])
+      expect(f).to receive(:xpath).with('./*[@duration]').and_return([video_data_el])
+      rf = described_class.new(double('Resource'), f, double('Rights'))
+      expect(Embed::MediaDuration).to receive(:new).and_call_original
+      expect(rf.duration).to eq '1:02:03'
+    end
+
+    it 'gets duration string from audioData' do
+      f = double('File')
+      audio_data_el = Nokogiri::XML("<audioData duration='PT43S'/>").root
+      expect(f).to receive(:xpath).with('./*/@duration').and_return(['something'])
+      expect(f).to receive(:xpath).with('./*[@duration]').and_return([audio_data_el])
+      rf = described_class.new(double('Resource'), f, double('Rights'))
+      expect(Embed::MediaDuration).to receive(:new).and_call_original
+      expect(rf.duration).to eq '0:43'
+    end
+
+    it 'nil when missing media data element' do
+      f = double('File')
+      allow(f).to receive(:xpath)
+      rf = described_class.new(double('Resource'), f, double('Rights'))
+      expect(Embed::MediaDuration).not_to receive(:new)
+      expect(rf.duration).to be_nil
+    end
+
+    it 'invalid format returns nil and logs an error' do
+      f = double('File')
+      audio_data_el = Nokogiri::XML("<audioData duration='invalid'/>").root
+      expect(f).to receive(:xpath).with('./*/@duration').and_return(['something'])
+      expect(f).to receive(:xpath).with('./*[@duration]').and_return([audio_data_el])
+      rf = described_class.new(double('Resource'), f, double('Rights'))
+      expect(Honeybadger).to receive(:notify).with("ResourceFile#media duration ISO8601::Errors::UnknownPattern: 'invalid'")
+      expect(Embed::MediaDuration).to receive(:new).and_call_original
+      expect(rf.duration).to be_nil
+    end
+  end
+
+  describe 'video_data' do
+    context 'valid videoData' do
```

Comment:
Can you use `with valid videoData` and then remove the exception from .rubocop_todo?